packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -2841,20 +2841,26 @@ export class ThinEngine extends AbstractEngine {
         source = InternalTextureSource.Unknown
     ): InternalTexture {
         let generateMipMaps = false;
+        let createMipMaps = false;
         let type = Constants.TEXTURETYPE_UNSIGNED_INT;
         let samplingMode = Constants.TEXTURE_TRILINEAR_SAMPLINGMODE;
         let format = Constants.TEXTUREFORMAT_RGBA;
         let useSRGBBuffer = false;
         let samples = 1;
         let label: string | undefined;
+        let createMSAATexture = false;
+        let comparisonFunction = 0;
         if (options !== undefined && typeof options === "object") {
             generateMipMaps = !!options.generateMipMaps;
+            createMipMaps = !!options.createMipMaps;

```

Comment:
What is the difference with generateMipMaps ?

Reply:
`createMipMaps` is used to create additional storage for the mipmaps when the texture is created, while `generateMipMaps` is used to indicate that we should generate the mipmaps when we unbind a MSAA framebuffer. It's possible to have `createMipMaps=true` and `generateMipMaps=false`, which means that the texture should be created with storage for mipmaps, but that mipmaps generation will be handled by the user and not automatically.